addons/hr_expense/tests/test_expenses.py

```diff
@@ -794,6 +793,76 @@ def test_payment_edit_fields(self):
 
         payment.write({'is_move_sent': True})
 
+    def test_corner_case_expense_reported_cannot_be_zero(self):
+        """
+        Test that the expenses are not submitted if the total amount is 0.0 nor able to be edited that way
+        unless unlinking it from the expense sheet.
+        """
+        expense = self.create_expense({'total_amount': 0.0, 'total_amount_currency': 0.0})
+
+        # CASE 1: FORBIDS Trying to submit an expense with a total_amount(_currency) of 0.0
+        with self.assertRaises(UserError):
+            expense.action_submit_expenses()
+
+        # CASE 2: FORBIDS Trying to change the total_amount(_currency) to 0.0 when the expense is linked to a sheet
+        expense.total_amount_currency = 1000
+        expense_sheet = expense._create_sheets_from_expense()
+        with self.assertRaises(UserError):
+            expense.total_amount_currency = 0.0
+        with self.assertRaises(UserError):
+            expense.total_amount = 0.0
+
+        # CASE 3: FORBIDS Trying to change the total_amount(_currency) to 0.0 when the expense sheet is approved
+        expense_sheet.action_approve_expense_sheets()
+        with self.assertRaises(UserError):
+            expense.total_amount_currency = 0.0
+        with self.assertRaises(UserError):
+            expense.total_amount = 0.0
+
+        # CASE 4: FORBIDS Trying to change the total_amount(_currency) to 0.0 when the expense sheet is posted and the account move created
+        expense_sheet.action_sheet_move_create()
+        with self.assertRaises(UserError):
+            expense.total_amount_currency = 0.0
+        with self.assertRaises(UserError):
+            expense.total_amount = 0.0
+
+        # CASE 5: Should behave like CASE 2 (FORBIDS)
+        expense_sheet.action_reset_expense_sheets()
+        with self.assertRaises(UserError):
+            expense.total_amount_currency = 0.0
+        with self.assertRaises(UserError):
+            expense.total_amount = 0.0
+
+        # CASE 6: ALLOWS Changing the total_amount(_currency) to 0.0 when the expense is unlinked from its sheet
+        expense.sheet_id = False
+        expense.write({'total_amount_currency': 0.0, 'total_amount': 0.0})
+
+    def test_corner_case_expense_prevent_empty_sheet_approval_actions(self):
+        """
+        Test that the expenses are not submitted or more if the sheet has no lines
```

Comment:
what does "more" mean?
I think the docstring should be absent because the name of the function is enough or it should be complete but in the current case, it's just a teaser :laughing: